crates/polars-plan/src/plans/conversion/dsl_to_ir.rs

```diff
@@ -664,8 +664,20 @@ pub fn to_alp_impl(lp: DslPlan, ctxt: &mut DslConversionContext) -> PolarsResult
 
             let subset = options
                 .subset
-                .map(|s| expand_selectors(s, input_schema.as_ref(), &[]))
+                .map(|s| {
+                    let cols = expand_selectors(s, input_schema.as_ref(), &[])?;
+
+                    // Checking if subset columns exist in the dataframe
+                    cols.iter()
```

Comment:
A bit nitpicky, but I think this can be written a little bit less verbose.

```rust
for c in &cols {
   let _ = input_schema.try_get(name)?
}

```

Reply:
Haha no problem. My understanding was, if the subset column(s) does not exist we need to raise `polars.exceptions.ColumnNotFoundError`. This approach would raise `polars.exceptions.SchemaFieldNotFoundError`. On a high level look both seem appropriate to me, but if `SchemaFieldNotFoundError` is correct, I can update this piece to less verbose like you suggested.

Something like this:
```rust
let cols = expand_selectors(s, input_schema.as_ref(), &[])?;

// Checking if subset columns exist in the dataframe
for col in cols.iter() {
    let _ = input_schema.try_get(col)?;
}

Ok::<_, PolarsError>(cols)
```

Reply:
We can `map_err`